src/Elastic.Markdown/Myst/Directives/Diagram/DiagramBlock.cs

```diff
@@ -47,7 +65,15 @@ public override void FinalizeAndValidate(ParserContext context)
 		catch (Exception ex)
 		{
 			this.EmitError($"Failed to encode diagram: {ex.Message}", ex);
+			return;
 		}
+
+		// Register diagram for tracking and cleanup
+		DiagramRegistry.RegisterDiagram(LocalSvgPath);
+
+		// Cache diagram asynchronously - fire and forget
+		// Use simplified approach without lock files to avoid orphaned locks
+		_ = Task.Run(() => TryCacheDiagramAsync(context));
```

Comment:
I think its best to `await` these and take the hit on build times. 

In most cases these will be cached now, we'd only add to the build time for legitimate updates. 


This could be done in `GenerateAll()` at the end e.g

```csharp
await diagramRegistry.CreateDiagramCachedFiles();
```

Which could then leverage `Parallel.ForEachAsyc()` to handle the updates and we can log information about outgoing calls in a more controlled fashion. E.g they always happen at the end of the build.